core/java/com/google/instrumentation/stats/View.java

```diff
@@ -32,15 +32,48 @@
   /**
    * Applies the given match function to the underlying data type.
    */
-  public abstract <T> T match(
-      Function<DistributionView, T> p0,
-      Function<IntervalView, T> p1);
+  public abstract <T> T match(Visitor<T> visitor);
 
   // Prevents this class from being subclassed anywhere else.
   private View() {
   }
 
   /**
+   * Visitor for the View type.
+   */
+  public abstract static class Visitor<T> {
+    public abstract T apply(DistributionView view);
+
+    public abstract T apply(IntervalView view);
+
+    /**
+     * Creates a Visitor with two {@link Function} objects.
+     *
+     * <p>Example:
+     *
+     * <pre>{@code
+     *   int numAggregations = myView.match(View.Visitor.create(
+     *       d -> d.getDistributionAggregations().size(),
+     *       i -> i.getIntervalAggregations().size()));
+     *  }</pre>
+     */
+    public static <T> Visitor<T> create(
```

Comment:
move this one level higher and it is a little less awkward. ex `View.createVisitor`